File: src/main/java/com/gisiona/demodevinia/domain/port/UserService.java
package com.gisiona.demodevinia.domain.port;

import com.gisiona.demodevinia.domain.model.User;
import java.util.List;

public interface UserService {
    
    User createUser(String name, String email);
    
    User getUserById(Long id);
    
    User getUserByEmail(String email);
    
    List<User> getAllUsers();
    
    User updateUser(Long id, String name, String email);
    
    void deleteUser(Long id);
}

File: src/main/kotlin/com/gisiona/demodevinia/domain/port/UserService.kt
package com.gisiona.demodevinia.domain.port

import com.gisiona.demodevinia.domain.model.User
import java.util.UUID

interface UserService {
    fun createUser(name: String, email: String): User
    fun getUserById(id: UUID): User?
    fun getUserByEmail(email: String): User?
    fun getAllUsers(): List<User>
    fun updateUser(id: UUID, name: String?, email: String?): User?
    fun deleteUser(id: UUID): Boolean
}
